src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -62,7 +62,12 @@ class VLMPipeline::VLMPipelineImpl : public VLMPipelineBase{
     size_t m_image_id = 0;
     size_t m_video_id = 0;
     ChatHistory m_history;
-
+    // if True, full history will be used as prompt on each chat generation
+    bool m_use_full_chat_history = false;
+    // It stores encoded images in case when m_use_full_chat_history is true
+    std::vector<ov::genai::EncodedImage> m_encoded_images;
```

Comment:
How about videos? If it is not supported with NPU, let's add OPENVINO_ASSERT when videos are passed to generate

Reply:
for the GenAI path the answer is not yet